src/libos/crates/async-io/src/util/channel.rs

```diff
@@ -300,6 +304,12 @@ impl File for Consumer {
     // TODO: implement readv
 
     fn poll(&self, mask: Events, poller: Option<&mut Poller>) -> Events {
+        if self.peer_end().is_shutdown() {
```

Comment:
Should we check the existing event before return Events::HUP?

Reply:
Done.